### PR TITLE
[+] hw6

### PR DESCRIPTION
**Pull request by dthmrphn**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -10,10 +10,25 @@ jobs:
...
+      DB_PASSWORD: 1234
+      DB_DSN: $DB_USER:$DB_PASSWORD@/$DB_NAME
+
     steps:
     - uses: actions/checkout@v3
 
+    - name: Set up Mysql
+      run : |
+          sudo /etc/init.d/mysql start
+          mysql -h 127.0.0.1 -e 'CREATE DATABASE ${{ env.DB_NAME }};' -u${{ env.DB_USER }} -p${{ env.DB_PASSWORD }}
```
Comment (uladzislaudd) [inline .github/workflows/go.yml]: I think there should be spaces here:
`-u**_**${{ env.DB_USER }}`
`-p**_**${{ env.DB_PASSWORD }}`

Comment (dthmrphn) [reply .github/workflows/go.yml]: didnt help

Comment (dthmrphn) [reply .github/workflows/go.yml]: so executing of this requires sudo and no space after -p flag

File: hw6_db_explorer/sample_db.sql [hunk 2]
```diff
@@ -32,4 +32,4 @@ CREATE TABLE `users` (
 INSERT INTO `users` (`user_id`, `login`, `password`, `email`, `info`, `updated`) VALUES
 (1,	'rvasily',	'love',	'rvasily@example.com',	'none',	NULL);
 
--- 2017-11-22 23:33:12
\ No newline at end of file
+-- 2017-11-22 23:33:12
```
Comment (uladzislaudd) [inline hw6_db_explorer/sample_db.sql:35]: what's going on here?

File: hw6_db_explorer/router_test.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		fmt.Fprintf(w, "records of table [%s] at [%s:%s]", table, offset, limit)
+	}
+
+	stubCreate := func(w http.ResponseWriter, r *http.Request) {
+		table := RouteParam(r, "table")
+		user_id := JSONParam(r, "user_id")
+		fmt.Fprintf(w, "creating record of table [%s] with 'user_id=%s'", table, user_id)
+	}
+
+	r := NewRouter()
```
Comment (uladzislaudd) [inline hw6_db_explorer/router_test.go:95]: take a look at [this](https://github.com/gorilla/mux)

File: hw6_db_explorer/router_test.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	srv := httptest.NewServer(http.HandlerFunc(r.Serve))
+	cli := &http.Client{}
+
+	for i, test := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			req, err := newRequest(test.method, srv.URL+test.path+test.query, test.body)
+			if err != nil {
+				t.Fatalf("request error at case[%s]: %s\n", t.Name(), err.Error())
```
Comment (uladzislaudd) [inline hw6_db_explorer/router_test.go:121]: take a look a [this package](https://github.com/stretchr/testify)

Comment (dthmrphn) [reply hw6_db_explorer/router_test.go:121]: ill take a look in future

File: hw6_db_explorer/sample_db.sql (hunk 2 above)
Comment (dthmrphn) [reply hw6_db_explorer/sample_db.sql:35]: something weird dunno

**Review by uladzislaudd — APPROVED** [review]
LGFM

**Pull request merged**